nutanix/resource_nutanix_foundation_central_image_nodes.go

```diff
@@ -497,17 +882,80 @@ func resourceNutanixFCImageClusterCreate(ctx context.Context, d *schema.Resource
 
 	if progress, ok := info.(*fc.ImagedClusterDetails); ok {
 		if utils.Float64Value(progress.ClusterStatus.AggregatePercentComplete) < float64(aggregatePercentComplete) {
-			return diag.Errorf("Progress is incomplete")
+			return collectIndividualErrorDiagnosticsFC(progress)
 		}
 	}
 
-	return nil
+	return resourceNutanixFCImageClusterRead(ctx, d, meta)
 }
 
 func resourceNutanixFCImageClusterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	return nil
 }
 
 func resourceNutanixFCImageClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).FC
+	log.Printf("[DEBUG] Deleting Cluster: %s, %s", d.Get("cluster_name").(string), d.Id())
+	err := conn.DeleteCluster(ctx, d.Id())
+	if err != nil {
+		if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error while Deleting Cluster: UUID(%s): %s", d.Id(), err)
+	}
+	d.SetId("")
 	return nil
 }
+
+func collectIndividualErrorDiagnosticsFC(progress *fc.ImagedClusterDetails) diag.Diagnostics {
+	// create empty diagnostics
+	var diags diag.Diagnostics
+
+	// append errors for failed node imaging
+	for _, v := range progress.ClusterStatus.NodeProgressDetails {
+		if utils.Float64Value(v.PercentComplete) < float64(aggregatePercentComplete) {
+			message := ""
+			for _, v1 := range v.MessageList {
+				message += *v1
+			}
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("Node imaging for imaged_node_uuid IP: %s failed with error:  %s.", *v.ImagedNodeUUID, *v.Status),
```

Comment:
`imaged_node_uuid IP` can we have something else as an identifier like cvm IP or something using which user can easily identify the node.